src/main/webapp/components/Fundings.js

```diff
@@ -111,11 +184,14 @@ export const Fundings = hh(class Fundings extends Component {
                       index: idx,
                       name: "source",
                       options: fundingOptions,
-                      value: this.props.fundings[idx].source,
+                      value: this.props.edit ? rd.future.source : rd.source,
+                      currentValue: this.props.edit ? current[idx].current.source : rd.source,
                       onChange: this.handleFundingSelect,
-                      error: this.props.error && idx === 0,
+                      error: this.props.edit === true  && this.props.errorIndex !== null? this.props.error && this.props.errorIndex.includes(idx) : this.props.error && idx === 0,
```

Comment:
To be 100% honest, these kinds of multi-conditional checks make my eyes cross. This seems like a good place to abstract the logic into a function that is easier to grok.

Reply:
This is now corrected, I must have overlooked it...